src/privileged/multipreffer/api.js

```diff
@@ -30,6 +32,9 @@ this.multipreffer = class extends ExtensionAPI {
 
 this.FirefoxHooks = {
   init() {
+    AddonStudies.addUnenrollListener(gExtension.id, async (reason) => {
+      await this.cleanup();
```

Comment:
Would it be fine if `cleanup` is called twice, once here and once in the normal uninstall hook?

Reply:
It did not cause any trouble during testing. Can you think of any issues with it?
I'm thinking it is preferable to leave both in if the user finds a way to uninstall themselves.

Reply:
I think it should be ok, though I'm a little unclear about this part: https://github.com/0c0w3/enhanced-tracking-protection-study/blob/d69429fbdd602cc61a559909c6e5b45b1b67cc2c/src/privileged/multipreffer/api.js#L109-L121

Reply:
ah, yes, understandably. What's happening there is we are flickering the cookie pref. It changes, then goes back to default. It would be fine if it flickers twice.
This is done to reset the `browser.contentblocking.category` pref. This is a unique pref that depends on the state of a handful of privacy prefs (we are working on a better solution for this pref, but in the meantime this is what we are working with). In order to ensure the user ends up back in the same state as they were originally in we need to clear the category pref, then reset (flicker) a pref it depends on to force it to reset itself appropriately. I can explain more thoroughly if you are curious.

Reply:
Wow, that's an exciting situation.

> It would be fine if it flickers twice.

I'll take your word for it. I don't need to know the more thorough explanation.